### DISCUSSION TITLE
3.8.0 not launching on Windows (VCRUNTIME140_1.dll not found)

### DISCUSSION THREAD

**Discussion by klepto84** [jamulussoftware/jamulus#1843] [Technical Support & Help] [answered]
Jamulus was working fine until I installed 3.8.  Now I get this message: The code execution cannot proceed because VCRUNTIME140_1.dll was not found. Reinstalling the program may fix this problem.       I have uninstalled, restarted and reinstalled 3 times.  Same message.  Any suggestions?  I am using Windows 10

**Comment by ann0see** [comment]
I assume something went wrong with the installer or your Windows install is broken. Did you try a new download or 3.7.0?

**Reply by klepto84** [reply]
Thanks for the quick reply. I tried the installer on Github and also Source Forge. Downloaded from both two times.  I will go back to 3.7

**Comment by hoffie** [comment]
Hrm, that's strange, there haven't been any deliberate changes in that area that I'd know of.
Can you confirm that you have the necessary [runtime](https://aka.ms/vs/16/release/vc_redist.x64.exe) (I think this is supposed to be handled by the installer, though)?

@jamulussoftware/maindevelopers I noticed that we do not use pinned versions in autobuild either (not sure whether it's possible at all, either). However, this might mean we've ended up building rc2 (for example) with a different toolchain than the final version, which would not be good... I haven't spotted any major updates for Windows Server 2019 virtual environments in the past days though (although a new image was released on 2021-05-31).

**Reply by klepto84** [reply]
Thanks. I used your runtime link and did a "repair". Downloaded and installed 3.8.0 same result.  Where can I find 3.7.0? All the links I've found go to 3.8.0

**Reply by klepto84** [reply]
I was using Jamulus for an hour when I noticed the update available message. I haven't been able to launch since installing 3.8.0

**Reply by hoffie** [reply]
> I noticed that we do not use pinned versions in autobuild either (not sure whether it's possible at all, either).

Looks like it isn't: https://github.com/actions/virtual-environments/issues/2505#issuecomment-764354728
The best thing we could do is use `windows-2019` instead of `windows-latest`. This wouldn't have helped here (even if the issue comes from a different base version).

**Reply by ann0see** [reply]
The installer _should_ handle everything automatically and does it on my PC & clean Windows 8.1 VM. So I think that it‘s a specific issue to @klepto84 ‘s setup.

**Comment by klepto84** [comment]
I have tried new downloads, but can't find 3.7 to download and install.

**Reply by pljones** [reply]
https://github.com/jamulussoftware/jamulus/releases/
You need to scroll down and then open up the "Assets".

**Comment by klepto84** [comment]
Thanks! Found it. Will stick with 3.7 for now.

Thanks again for your help.  I just tested and my
midi-Focusright-Reaper setup is working again!

**Comment by klepto84** [comment]
Thanks again guys for your assistance.
Going back to 3.7.0 was fine.  I just had a 90 minute session with a buddy
with no issues.
I used my midi keyboard, bass, and mic during the session.

I am using the Reaper reroute function to send the 3 midi and audio signals
to Jamulus. I can also send recorded backing tracks through Reaper and play
another instrument along with my vocal.

I get the system error message at the end of the install process, when
Jamulus 3.8.0 should open.  At that time, my midi keyboard is turned off
and Reaper is not active.
So I don't think my audio routing is impacting the situation. I updated
Runtime as per the previous message.  Can you suggest anything else I
should try?  I don't want to be stuck in 3.7.0 time warp forever as I
expect Jamulus is here to stay and will only get better!

Thanks again!

**Reply by ann0see** [reply]
Yes sure! You shouldn’t need to stay on an old version. 

I still have the impression that something is wrong with your OS/Installation. 
Could you please take a screenshot of the Jamulus folder im program files?

Maybe we can compare if all necessary files are there. 

Did you also try to remove and install the runtime again? 
One option (which is always the last option and is mit preferred!!!) might be an implace update of your OS (with all your data and programs backed up before doing so). This might install missing files again.

**Comment by klepto84** [comment]
Here you go.

**Reply by ann0see** [reply]
Could you please upload pictures on the web interface  (https://github.com/jamulussoftware/jamulus/discussions/1843)? I can’t see anything since you sent an e-mail.

**Reply by klepto84** [reply]
![Screenshot (32)](https://user-images.githubusercontent.com/76572781/121165829-8c19b080-c81e-11eb-8487-4a895d904a25.png)

**Comment by klepto84** [comment]
I uploaded the screenshot to Github, but the interface looks like 1980's MS
DOS and I'm not sure if you will be able to open the file.

**Reply by henkdegroot** [reply]
i have noticed this issue myself as well on 1 of my Win10 machines. I resolved this by manually removing the "Microsoft Visual C++ 2015-2019 Redistributable (x64) - 14.29.30037" from my installed programs. After this downloaded it from microsoft and manually installed and now it is working.

**EDIT**: Link to microsoft to download: [The latest supported Visual C++ downloads](https://support.microsoft.com/en-us/topic/the-latest-supported-visual-c-downloads-2647da03-1eea-4433-9aff-95f26a218cc0)
Download the one for Visual Studio 2015, 2017 and 2019 which matches your platform

Jamulus (and possible other programs) just can't find the VCRUN...dll in any of the paths windows is aware of. After removing and installing again, the dll is added back into the system32 folder. On my system it was no longer in that folder. Other programs failed to work as well on my system.

The repair method in the vc-runtime, seems to not work correct and when jamulus is installed and finds it is already in the installed programs, it will skip the installation.

More investigation to be done to find the actual cause....but at least wanted to share a way to resolve the issue and use 3.8.0.

**Reply by pcar75** [reply]
@henkdegroot : 
Bonjour, I'm thinking of waiting a bit before migrating (from 3.6.2) to 3.8.x ...
1. Two players also mentioned problems with Windows 10 and 3.8.0 ,  but one of them was also trying to connect to Reaper through ReaRoute (with Jamulus and a Focusrite audio interface + ASIO4ALL ?! )
2. Yesterday, our drummer easily installed 3.8.0, but on his Windows 7Pro (64b) laptop (no DAW or MIDI connection)
3. In case VCRUNTIME*.dll is a problem , I wouldn't  know where to put it.  On my Windows 10 -regularly updated-
- A CMD/DOS 'dir' shows 33 occurences of   VCRUNTIME*.DLL
- An Explorer 'search' shows 35 of               *VCRUNTIME*.DLL
- Most have different file size and 32b and 64b  are peppered all over my program drive including \Windows\System32, \Windows\SysWOW64, \Windows\WinSxS, \Program Files and \Program Files (86), ...
(Aarrgghh ! don't you love bloated software :-/  ... I haven't worked up the courage yet to low-format and reinstall everything (probably weeks) .

So what would be the best directory to put in in ?

**Reply by henkdegroot** [reply]
@pcar75 , I don't believe you will have much luck if you just copy the file into any folder. In case you upgrade and get the message that the VCRUNTIME140.dll is not found, you should be able to resolve that be removing the "Microsoft Visual C++ 2015-2019 Redistributable (x64) - 14.29.30037" and then install it again using the vc_redist.x64.exe install from Microsoft. This should do a proper installation. 

I think it would be good to know if you performed an upgrade and got the issue or if you manually removed the previous jamulus version and then installed the latest. As we are not sure what is the cause. I have just checked and do see that the runtime provided with 3.7.0 is a bit older version compared to the one provided with 3.8.0.

**Reply by ann0see** [reply]
> or if you manually removed the previous jamulus version and then installed the latest. 

The installer should run the uninstaller of the previous installation before starting the install process.

> with 3.7.0 is a bit older version compared to the one provided with 3.8.0.

Interesting... Probably that's the issue then?

**Reply by henkdegroot** [reply]
> Interesting... Probably that's the issue then?

Does not seem to be it....in the meanwhile I installed several version and uninstalled and upgrade and etc.....on the same machine as I previously had it broken.

Currently thinking perhaps something with a dll (from the runtime dlls) is in use by another program and does allow to be updated. Again that all should normally be handled by windows...but got to start somewhere. All logic in jamulus for installation/uninstall/upgrade seem to be working oke.

**Reply by pcar75** [reply]
I just Installed 3.8.0 over 3.6.2.  No problem installing.

However I can't understand why there is now so much noise (gurgling) when connecting to 127.0.0.1 and the Jitter buffer (auto) level shoots to 11 local ( server level = 2 ; these were around 2 and 2 with 3.6.2 ) ? 
Jamulus and ASIO settings are unchanged  and a short test with a public server seems as good as before.

*Should I open another Discussion topic ?*  EDIT : Done (#1855

**Reply by henkdegroot** [reply]
> Should I open another Discussion topic ?

That would make sense, as it is not related to this issue. FYI, I do not have this experience myself..but it sounds strange indeed.

**Reply by pcar75** [reply]
> 
> 
> @pcar75 , I don't believe you will have much luck if you just copy the file into any folder. In case you upgrade and get the message that the VCRUNTIME140.dll is not found, you should be able to resolve that be removing the "Microsoft Visual C++ 2015-2019 Redistributable (x64) - 14.29.30037" and then install it again using the vc_redist.x64.exe install from Microsoft. This should do a proper installation.
> 
> I think it would be good to know if you performed an upgrade and got the issue or if you manually removed the previous jamulus version and then installed the latest. As we are not sure what is the cause. I have just checked and do see that the runtime provided with 3.7.0 is a bit older version compared to the one provided with 3.8.0.

After re-installing 3.6.2 , testing and re-installing 3,8,0 , Jamulus can't launch , giving a different missing Windows components (MSVCP140_1.dll) , Ref. `(#1855)`

**Reply by pljones** [reply]
- Go to "Add or remove programs"
- Find "Microsoft Visual C++ 2015-2019 Redistributable (x64)" (any version)
- Uninstall it

Repeat until there are no versions of exactly that Redistributable installed.  I had two at one point.

Then use the 3.8.0 installer.  That then worked for me.

**Reply by pcar75** [reply]
@pljones 
Thanks but some more details please .

You may have seen in `#1855` that CONCRT140.dll is now missing for Bandlab's Cakewalk !

1. Is that also part of "Microsoft Visual C++ 2015-2019 Redistributable (x64)" ?

2. Do I need to uninstall 3.8.0 first ? 
*Note that the last 3.8.0 installer attempt did not take well to having a C:\Program Files\Jamulus already present !*

3. Will the Jamulus installer re-install a current version of "Microsoft Visual C++ 2015-2019 Redistributable (x64)" in Windows directories or must I install it myself from a Windows site (which one) ?

4. Are there reboots necessary between some (each?) steps .

Thanks in advance.

**Reply by henkdegroot** [reply]
@pcar75 , CONCRT140.dll is also part of the Microsoft Visual C++ 2015-2019 Redistributable x64.
3.8.0 should be uninstalled before installing it again.
Also make sure you have removed the VC++ using the "add or remove programs". This is important as otherwise the installation (part of the jamulus install) will not be done correct as it believes it is already installed.
Next run the 3.8.0 jamulus installer will then indeed re-install the VC++ redist and should install in the "regular" windows folders.

I have not experienced yet that I need to reboot between any of these steps.

**EDIT:** Forgot to mention, you can remove (or rename) the Jamulus folder in the Program Files folder before installing jamulus.

**Reply by pcar75** [reply]
Thanks All !

Just to be on the safe side, I ran MS support suggestions in a CMD admin window :
- DISM.exe /Online /Cleanup-image /Restorehealth
It told me that the restore operation completed successful.  Then
- sfc /scannow
It told me that there were come corrupted files and they were repaired, 
but I still had the Jamulus and Cakewalk errors after a clean reboot.  

Then I
- Uninstalled Jamulus 3.8.0,
- Uninstalled "Microsoft Visual C++ 2015-2019 Redistributable (x64)"
- Reinstalled Jamulus 3.8.0
- Rebooted

Now , neither Jamulus nor Cakewalk miss anything ; that problem solved !
(for the moment anyway ...  as MS recently announced a Windows 11 version for later this year :-/

**Comment by klepto84** [comment]
Thanks!

**Comment by klepto84** [comment]
First of all I am the Reaper guy and to make sure we have our facts strait.
I have been routing my midi keyboard M-Audio Hammer88 Pro, and both
channels of my Scarlette through Reaper to Jamulus with no issues. I also
have a backing track channel and multiple midi instruments successfully
ASIO4ALL is not part of the set up.  It was working flawlessly until I
installed 3.8.0. For the record Reaper was not involved in the install and
the keyboard was powered down.  My routing has nothing to do with this
issue. So to be clear.  I am not "trying to connect.....etc" I have been
connecting.
I was unable to launch Jamulus 3.8.0 so I never tested the routing.  I
reinstalled 3.7.0 and my Reaper routing is working great!

I plan to hang with 3.7.0 until the bugs are worked out and try again in a
few months.

**Reply by pcar75** [reply]
@klepto84 
Oopps ! Sorry for the misunderstanding !

**Reply by henkdegroot** [reply]
> I plan to hang with 3.7.0 until the bugs are worked out and try again in a few months.

It should not be a problem at all for you to stick with that version, when it works fine for you. At the moment I don't think this is a Jamulus problem but more a Windows related issue. When I hit the issue, I had previously removed many programs, so not sure which program made my "visual studio runtime" corrupt. Not all programs are as nice when it comes to uninstalling and they remove items they should not. Jamulus only installs/upgrades the runtime during the installation and is not removing it when Jamulus is removed.

It would be good to know if your issue can also be fixed by doing a manual remove/install of the visual studio runtime. Even the re-install of Jamulus may have resolved that issue as well and the next upgrade to 3.8.0 might be without any issue.
Any other information there is share my help finding the reason why this happened and once that is known it might be possible to find a more permanent solution for this.

**Comment by klepto84** [comment]
No worries!

**Reply by pcar75** [reply]
Bonjour @klepto84 ,
Could you please test 3.7.0 to see if you get similar, better or worse performance  /soundquality with your loopback (127.0.0.1) server  than with known public ones.
As per #1855, I wish to know if I should downgrade to 3.7.0 or to my initial 3.6.2.
Thanks in advance.

**Comment by pljones** [comment, marked as answer]
I tried to pin this down earlier this evening.

My installs had been of release artifacts - not from the autobuilds.

I had no problem with 3.8.0 release.

I had a problem trying out the PortAudio autobuild.

Re-installing 3.8.0 releases didn't fix it - I had to manually uninstall the VC++ Redistributable and then install 3.8.0 again.  At that time, installing the PortAudio build immediately caused the problem again.

When I tried to pin the issue down, however... I couldn't reproduce it reliably.  I had reverted to 3.8.0, tried the PortAudio build, got the error, tried other builds without success.  Then uninstalled the VC++ Redistributable and installed 3.8.0 to confirm that still worked.  Then the latest PortAudio build worked, too, when overwriting it.

What I've not tried again is
- Uninstall VC++ Redistributable
- Install 3.8.0 Release to C:\Program Files\Jamulus or wherever normally installed
- Confirm it works
- Install some other build from the autobuild artifacts **to a new location**
- Confirm it's broken
- **Repeat**

It may be some intermittent download error.  I've had a Windows Update that's been failing for ages lately:
- June 11, 2021—KB5004476 (OS Builds 19041.1055, 19042.1055, and 19043.1055) Out-of-band
Last failed install attempt on ‎17/‎06/‎2021 - 0x800f0831

whereas
- June 8, 2021—KB5003637 (OS Builds 19041.1052, 19042.1052, and 19043.1052)
Successfully installed on ‎09/‎06/‎2021

**Comment by ann0see** [comment]
Had a similar problem with a choir member I helped which could be solved by following [the instructions by @pcar75](https://github.com/jamulussoftware/jamulus/discussions/1843#discussioncomment-940569)

> Thanks All !

> Just to be on the safe side, I ran MS support suggestions in a CMD admin window :
> - DISM.exe /Online /Cleanup-image /Restorehealth
> It told me that the restore operation completed successful.  Then
> - sfc /scannow
> It told me that there were come corrupted files and they were repaired, 
> but I still had the Jamulus and Cakewalk errors after a clean reboot.  

> Then I
> - Uninstalled Jamulus 3.8.0,
> - Uninstalled "Microsoft Visual C++ 2015-2019 Redistributable (x64)"
> - Reinstalled Jamulus 3.8.0
> - Rebooted

> Now , neither Jamulus nor Cakewalk miss anything ; that problem solved !
> (for the moment anyway ...  as MS recently announced a Windows 11 version for later this year :-/

**Reply by pljones** [reply]
I've since run one of the Microsoft updater clean up tools (dsim and sfc not having resolved my MS update issue) and not had the problem since, so I think it might have been a combination of a duff install of "Microsoft Visual C++ 2015-2019 Redistributable (x64)" from somewhere.

**Comment by rtrial** [comment]
I am told that there could be an issue with the link below.  I did not have an issue with it.  Fix #3 did work for me.  You can avoid the link by following the directions at the bottom of this message.

I had this same problem with 3.9.  The solution listed as Fix #3 in this post: https://techloris.com/vcruntime140dll-is-missing/ (Repair Microsoft Visual C++ Redistributed) worked for me.

Follow these directions and you will not need to go to the link.  The link suggests (in Windows 11) run Control Panel | Programs | Programs and Features | Right-click the Microsoft Visual C++ Redistributable appropriate to your system (32 or 64 bit OS setting) and Select Change | Repair.

**Reply by ann0see** [reply]
Just as comment, I'd skip the Restoro tip as it sounds like an ad for me. The app sounds a bit sketchy to me. 

https://davescomputertips.com/is-restoro-legitimate/

https://www.trustpilot.com/review/restoro.com

https://answers.microsoft.com/en-us/windows/forum/all/is-restoro-safe/fa70bdfb-9a4d-485f-96e8-1f1636ce298b

**Reply by pljones** [reply]
> rtrial joined yesterday

... just to post a link to a site for ResToro... yes, the post here does sound like an ad - I'd be inclined to delete.

**Reply by ann0see** [reply]
No, I wouldn't delete it as @rtrial can also be a normal person. We must stay transparent and not abuse admin powers. Rtrial doesn't promote option 1 which is the program in question. They have just found a somewhat sketchy site. So my suggestion would be that @rtrial, you edit your message and add a disclaimer that the link might lead to a biased article sounding like an ad. If you don't with a reasonable time, we'll add one. 

If the tip works, it's still ok.

**Reply by ann0see** [reply]
@rtrial thanks for your edit.

**Reply by ann0see** [reply]
Could you please put the disclaimer before the link?

... this post: -> the following post. Disclaimer: I was told ... do not want to click the link follow the instructions below. -link-

**Reply by ann0see** [reply]
Thanks again for adding the disclaimer.
